File: Java/src/main/java/com/xurxodev/stringcalculator/rule/NumberGreaterThanIgnoredRule.java
package com.xurxodev.stringcalculator.rule;

import com.xurxodev.stringcalculator.NumberRemoverByIgnoredRules;

public class NumberGreaterThanIgnoredRule implements NumberRemoverByIgnoredRules.IgnoredRule {

    private final Integer maxValue;

    public NumberGreaterThanIgnoredRule(Integer maxValue){
        this.maxValue = maxValue;
    }

    @Override
    public boolean isIgnored(Integer number) {
        return number > maxValue;
    }
}

File: Swift/stringcalculator/NumberGreaterThanIgnoredRule.swift
//
//  NumberGreaterThanIgnoredRule.swift
//  stringcalculator
//
//  Created by Jorge Sánchez on 13/10/16.
//  Copyright © 2016 xurxodev. All rights reserved.
//

import Foundation

class NumberGreaterThanIgnoredRule: IgnoredRule {
    
    let maxValue:Int
    
    init(maxValue:Int) {
        self.maxValue = maxValue
    }
    
    func isIgnored(number: Int) -> Bool{
        return number > maxValue        
    }
}